agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/wascore/statsbeat/AzureMetadataService.java

```diff
@@ -94,31 +93,43 @@ void parseJsonResponse(String response) throws IOException {
   public void run() {
     HttpRequest request = new HttpRequest(HttpMethod.GET, ENDPOINT);
     request.setHeader("Metadata", "true");
+    HttpResponse response;
     try {
-      HttpResponse response = LazyHttpClient.getInstance().send(request).block();
-      if (response == null) {
-        // this shouldn't happen, the mono should complete with a response or a failure
-        throw new AssertionError("http response mono returned empty");
-      }
-      parseJsonResponse(response.toString());
-    } catch (JsonEncodingException ex) {
-      // When it's not VM/VMSS, server does not return json back, and instead it returns text like
-      // the following:
-      // "<br />Error: NetworkUnreachable (0x2743). <br />System.Net.Sockets.SocketException A
-      // socket operation was attempted to an unreachable network 169.254.169.254:80".
-      logger.debug(
-          "This is not running from an Azure VM or VMSS. Shut down AzureMetadataService scheduler.");
-      scheduledExecutor.shutdown();
+      response = LazyHttpClient.getInstance().send(request).block();
     } catch (Exception ex) {
       if (ex.getCause()
```

Comment:
with this change, i think we don't need to know the cause.. because it will get shutdown by any exception assuming caused by the endpoint?